tools/chapel-py/src/method-tables/uast-methods.h

```diff
@@ -640,6 +640,16 @@ CLASS_BEGIN(VarLikeDecl)
                const char*, return intentToString(node->storageKind()))
 CLASS_END(VarLikeDecl)
 
+CLASS_BEGIN(Formal)
+  PLAIN_GETTER(Formal, is_this, "Check if this Formal node is a 'this' formal",
+               bool,
+                auto parent = chpl::parsing::parentAst(context, node);
+                return parent && parent->isFunction() &&
+                       parent->toFunction()->thisFormal() &&
+                       parent->toFunction()->thisFormal()->id() == node->id();
+              )
```

Comment:
`parentAst` is notoriously expensive. Have you found issues with literally checing if `name == USTR("this")`?

Reply:
I don't think there is any issue with that, since the parser prevents it from being used in any `Formal` name context